File: emailsender/languages/java/RelayPolicy.java
package io.amerhwitat.emailsender;

public record RelayPolicy(boolean authorized, boolean openRelay, boolean spoofFrom, boolean privacyMode) {
    public void validate() {
        if (openRelay || spoofFrom) throw new IllegalArgumentException("open relays and From spoofing are disabled");
        if (!authorized) throw new IllegalArgumentException("relay authorization is required");
    }
}

File: emailsender/languages/kotlin/RelayPolicy.kt
package io.amerhwitat.emailsender

data class RelayPolicy(val authorized: Boolean = true, val openRelay: Boolean = false, val spoofFrom: Boolean = false, val privacyMode: Boolean = false) {
    fun validate() {
        require(!openRelay && !spoofFrom) { "Open relays and From spoofing are disabled" }
        require(authorized) { "Relay authorization is required" }
    }
}
